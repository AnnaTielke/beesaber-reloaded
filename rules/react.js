module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
