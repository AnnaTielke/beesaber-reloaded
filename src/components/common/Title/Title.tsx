// import { useState } from 'react';
import styled from 'styled-components';

import NeonText from '../NeonText/NeonText';

import type { ColorStrings } from '@/tokens/definitions/color';
import type { ElementType } from '../NeonText/NeonText';

const Container = styled.div`
  display: flex;
  gap: 0.5rem;
  letter-spacing: 0.3rem;
`;

type Props = {
  as?: ElementType;
  children: string;
};

const Title = ({ children, as = 'h1' }: Props): JSX.Element => {
  const wordsArr = children.split(' ');
  const renderWords = (words: string[]) => {
    const colors: ColorStrings[] = ['red', 'blue'];

    return words.map((word, i) => (
      <NeonText
        key={word + i.toString()}
        glow
        as={as}
        titleColor={colors[i % 2]}
      >
        {word}
      </NeonText>
    ));
  };

  return <Container>{renderWords(wordsArr)}</Container>;
};

export default Title;
