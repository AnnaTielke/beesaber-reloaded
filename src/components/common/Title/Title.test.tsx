import { render } from '@testing-library/react';
import React from 'react';

import Title from './Title';

describe('Title', () => {
  it.each(['h1', 'h2', 'h3', 'p', 'span', undefined] as const)(
    'should match the snapshot when element is %s',
    (as) => {
      const { container } = render(<Title as={as}>Glowing Text</Title>);

      expect(container.firstChild).toMatchSnapshot();
    }
  );
});
