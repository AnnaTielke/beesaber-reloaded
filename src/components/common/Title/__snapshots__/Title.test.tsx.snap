// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match the snapshot when element is h1 1`] = `
.c1 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c2 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Glowing
  </h1>
  <h1
    class="c2"
  >
    Text
  </h1>
</div>
`;

exports[`Title should match the snapshot when element is h2 1`] = `
.c1 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c2 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    Glowing
  </h2>
  <h2
    class="c2"
  >
    Text
  </h2>
</div>
`;

exports[`Title should match the snapshot when element is h3 1`] = `
.c1 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c2 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    Glowing
  </h3>
  <h3
    class="c2"
  >
    Text
  </h3>
</div>
`;

exports[`Title should match the snapshot when element is p 1`] = `
.c1 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c2 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Glowing
  </p>
  <p
    class="c2"
  >
    Text
  </p>
</div>
`;

exports[`Title should match the snapshot when element is span 1`] = `
.c1 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c2 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Glowing
  </span>
  <span
    class="c2"
  >
    Text
  </span>
</div>
`;

exports[`Title should match the snapshot when element is undefined 1`] = `
.c1 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c2 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Glowing
  </h1>
  <h1
    class="c2"
  >
    Text
  </h1>
</div>
`;
